### PR TITLE
Relax Gamma Sampler types

### PR DESCRIPTION
**Pull request by dinarior**
* Relax T into Real in gamma sampler

### CODE REVIEW

**Comment by dinarior** [comment]
Fix for this:
https://github.com/JuliaStats/Distributions.jl/issues/956

**Comment by mschauer** [comment]
I think this might come at performance cost because functions with `GammaGDSampler` will not dispatch on the type parameter `T` anymore.

**Comment by dinarior** [comment]
Changed the ctor instead to have explicit casting to the correct type for all params (and not just some of them, as previously).

**Comment by mschauer** [comment]
Yes, this is better (Ctor = constructor I guess)

File: src/samplers/gamma.jl [hunk 1]
```diff
@@ -54,7 +54,7 @@ function GammaGDSampler(g::Gamma{T}) where {T}
         c = 0.1515/s
     end
 
-    GammaGDSampler(a,s2,s,i2s,d,q0,T(b),T(σ),T(c),scale(g))
+    GammaGDSampler(T(a),T(s2),T(s),T(i2s),T(d),T(q0),T(b),T(σ),T(c),scale(g))
```
Comment (matbesancon) [inline src/samplers/gamma.jl]:
```suggestion
    GammaGDSampler(T(a), T(s2), T(s), T(i2s), T(d), T(q0), T(b), T(σ), T(c), scale(g))
```

**Comment by matbesancon** [comment]
Thanks for this

**Pull request merged**